src/k8s/pkg/k8sd/pki/load.go

```diff
@@ -19,29 +19,59 @@ func loadCertificate(certPEM string, keyPEM string) (*x509.Certificate, *rsa.Pri
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to parse certificate: %w", err)
 	}
+	if keyPEM == "" {
+		return cert, nil, nil
+	}
 
-	var key *rsa.PrivateKey
-	if keyPEM != "" {
-		pb, _ := pem.Decode([]byte(keyPEM))
-		switch pb.Type {
-		case "RSA PRIVATE KEY":
-			key, err = x509.ParsePKCS1PrivateKey(pb.Bytes)
-			if err != nil {
-				return nil, nil, fmt.Errorf("failed to parse RSA private key: %w", err)
-			}
-		case "PRIVATE KEY":
-			parsed, err := x509.ParsePKCS8PrivateKey(pb.Bytes)
-			if err != nil {
-				return nil, nil, fmt.Errorf("failed to parse private key: %w", err)
-			}
-			v, ok := parsed.(*rsa.PrivateKey)
-			if !ok {
-				return nil, nil, fmt.Errorf("not an RSA private key")
-			}
-			key = v
-		default:
-			return nil, nil, fmt.Errorf("unknown private key block type %q", pb.Type)
-		}
+	key, err := LoadRSAPrivateKey(keyPEM)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to load RSA private key: %w", err)
 	}
+
 	return cert, key, nil
 }
+
+// LoadRSAPrivateKey parses the specified PEM block and return the rsa.PrivateKey.
+func LoadRSAPrivateKey(keyPEM string) (*rsa.PrivateKey, error) {
```

Comment:
We'll also need some tests for this. Feel free to add is as a task to the already created card.